### PR TITLE
Remove email and recovery related identity endpoints

### PR DESCRIPTION
**Pull request by jsdt**
# Description of Changes

This removes code for handling emails and identity recovery, since this should move elsewhere.

# API and ABI breaking changes

If this is an API or ABI breaking change, please apply the
corresponding GitHub label.

# Expected complexity level and risk

2. This is not very complex. We are just removing code. The risk here is in missing something that depends on these endpoints.

# Testing

This is failing the internal tests, but the internal `cargo test`s pass with [this PR](https://github.com/clockworklabs/SpacetimeDBPrivate/pull/1041).

### CODE REVIEW

**Comment by cloutiertyler** [comment]
This generally looks good to me. It's correct, although it will need a corresponding internal PR.

File: crates/client-api/src/auth.rs [hunk 1]
```diff
@@ -96,7 +97,15 @@ pub struct SpacetimeAuth {
 impl SpacetimeAuth {
     /// Allocate a new identity, and mint a new token for it.
     pub async fn alloc(ctx: &(impl NodeDelegate + ControlStateDelegate + ?Sized)) -> axum::response::Result<Self> {
-        let identity = ctx.create_identity().await.map_err(log_and_500)?;
+        // TODO: I'm just sticking in a random string until we change how identities are generated.
```
Comment (cloutiertyler) [inline crates/client-api/src/auth.rs:100]: We'll just have to remember to go back and swap this.

**Review by cloutiertyler — APPROVED** [review]
This LGTM

File: crates/client-api/src/auth.rs (hunk 1 above)
Comment (jsdt) [reply crates/client-api/src/auth.rs:100]: This will be fixed in the next PR.

**Pull request merged**